src/collector.py

```diff
@@ -7,35 +7,53 @@
 
 from aiogram import Bot
 from aiogram.types import Gift, Gifts
+import logging
 
-BASE_OUTPUT_DIR = Path("/tmp/gifts")
+BASE_OUTPUT_DIR = Path(getenv("OUTPUT_DIR", "/tmp/gifts"))
 IMAGES_DIR = BASE_OUTPUT_DIR.joinpath("images")
 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
 def prepare_dirs():
+    """
+    Create necessary directories and copy static files.
+    """
     for output_dir in (BASE_OUTPUT_DIR, IMAGES_DIR):
         if not output_dir.exists():
-            output_dir.mkdir()
-    for filename in ("index.html", "style.css"):
+            output_dir.mkdir(parents=True, exist_ok=True)
+    for filename in ("index.html", "style.css", "ru.index.html", "uk.index.html", "uz.index.html"):
         shutil.copyfile(src=f"web/{filename}", dst=BASE_OUTPUT_DIR.joinpath(f"{filename}"))
 
 def update_date():
-    html_file = BASE_OUTPUT_DIR.joinpath("index.html")
-    current_date = datetime.now(tz=timezone.utc).strftime("%Y-%m-%d %H:%M UTC")
-    with open(html_file, "r") as f:
-        content = f.read()
+    """
+    Update the date in the HTML file to the current date and time.
+    """
+    for filename in ("index.html", "ru.index.html", "uk.index.html", "uz.index.html"):
+        html_file = BASE_OUTPUT_DIR.joinpath(filename)
+        current_date = datetime.now(tz=timezone.utc).strftime("%Y-%m-%d %H:%M UTC")
+        with open(html_file, "r") as f:
+            content = f.read()
 
-    updated_content = content.replace("%DATE%", current_date)
-    with open(html_file, "w") as f:
-        f.write(updated_content)
+        updated_content = content.replace("%DATE%", current_date)
+        with open(html_file, "w") as f:
+            f.write(updated_content)
 
+async def download_gifts(bot: Bot):
+    """
+    Download available gifts and save their data and images.
 
-async def download_gifts(
-        bot: Bot,
-):
-    data = list()
+    Parameters:
+    bot (Bot): An instance of the aiogram Bot.
+    """
+    data = []
+    try:
+        available_gifts: Gifts = await bot.get_available_gifts()
+    except Exception as e:
+        logger.error(f"Failed to fetch available gifts: {e}")
+        return
```

Comment:
In case we cannot get available gifts, better crash GitHub Action, rather than silently fail